### PR TITLE
Extract aruba sw

### PR DESCRIPTION
**Pull request by benper44**

### CODE REVIEW

File: ExtractArubaSW.ps1 [hunk 1]
```diff
@@ -0,0 +1,683 @@
+﻿
+ <#
+  .SYNOPSIS
+  Ce script permet d'extraire des données d'un switch Aruba et de classer ces informations d'un tableau Excel
```
Comment (alagoutte) [inline ExtractArubaSW.ps1]: Can you translate to english your comment ?

File: ExtractArubaSW.ps1 [hunk 2]
```diff
@@ -0,0 +1,683 @@
...
+                                -Revision matérielle
+                                -Modèle du switch
+
+                                -Nom du port
+                                -Identifiant du port
+                                -Vlan ports taguer
+                                -Vlan ports non taguer
+                                -Statut LACP
+
+Ce script nécessite au le module PowerArubaSW soit installé. Plus d'information sur https://github.com/alagoutte/PowerArubaSW
```
Comment (alagoutte) [inline ExtractArubaSW.ps1]: use https://github.com/PowerAruba/PowerArubaSW

File: ExtractArubaSW.ps1 [hunk 3]
```diff
@@ -0,0 +1,683 @@
...
+
+if (Get-Module -ListAvailable -Name PowerArubaSW) {
+
+        Import-Module PowerArubaSW
+        cls
+
+} else {
+
+        cls
+        echo ""
```
Comment (alagoutte) [inline ExtractArubaSW.ps1]: avoid alias use Write-Host (and can be a only a line... no need all "" ##...

File: ExtractArubaSW.ps1 [hunk 4]
```diff
@@ -0,0 +1,683 @@
...
+    $resultarray += $item
+
+}
+
+
+$lldpremote = get-arubaswlldpremote
+
+$resultarray2=@()
+
+foreach ($resultarra in $resultarray) {
```
Comment (alagoutte) [inline ExtractArubaSW.ps1]: use the same loop for lldp ?

File: ExtractArubaSW.ps1 [hunk 5]
```diff
@@ -0,0 +1,683 @@
...
+    $item | Add-Member -type NoteProperty -Name 'lldp_port_description' -Value $($w.port_description)
+     
+    $resultarray2 += $item
+
+}
+
+# Creation du fichier Excel
+
+
+$infoswitch = Get-ArubaSWSystemStatus | Select -Property name,serial_number,firmware_version,hardware_revision,product_model
```
Comment (alagoutte) [inline ExtractArubaSW.ps1]: Need to add stacked switch support ;-)

File: ExtractArubaSW.ps1 [hunk 6]
```diff
@@ -0,0 +1,683 @@
...
+$Column = 1
+$anum = 1
+$bnum = 1
+
+
+
+
+# Saisie des données dans Excel
+
+ function excelparamarray1  {
```
Comment (alagoutte) [inline ExtractArubaSW.ps1]: better name ?

Comment (alagoutte) [reply ExtractArubaSW.ps1]: May be also use function for get info and also write a line on excel ?

File: ExtractArubaSW.ps1 [hunk 7]
```diff
@@ -0,0 +1,683 @@
...
+$bnum++ 
+
+
+$row = 1
+$Column = 3
+$anum = 1
+$bnum = 1
+
+# Récupération des données
+$entries = $infoswitch | Select -Property name,serial_number,firmware_version,hardware_revision,product_model
```
Comment (alagoutte) [inline ExtractArubaSW.ps1]: no need Select

File: ExtractArubaSW.ps1 [hunk 8]
```diff
@@ -0,0 +1,683 @@
...
+                                $sheet.Cells.Item($row,$column).Font.Name = "Cambria"
+                                $sheet.Cells.Item($row,$column).Interior.ColorIndex = 19
+                                $sheet.Cells.Range("C$($anum):G$($bnum)").Borders.LineStyle = [Microsoft.Office.Interop.Excel.XlLineStyle]::xlContinuous
+                                $sheet.Cells.Range("C$($anum):G$($bnum)").Borders.Weight = 3
+                          
+                            }
+
+
+
+foreach ($entry in $entries)  {
```
Comment (alagoutte) [inline ExtractArubaSW.ps1]: really need a loop ?

File: ExtractArubaSW.ps1 (hunk 1 above)
Comment (benper44) [reply ExtractArubaSW.ps1]: done

File: ExtractArubaSW.ps1 (hunk 2 above)
Comment (benper44) [reply ExtractArubaSW.ps1]: done

File: ExtractArubaSW.ps1 (hunk 3 above)
Comment (benper44) [reply ExtractArubaSW.ps1]: Not completely done, write-host ok

File: ExtractArubaSW.ps1 (hunk 4 above)
Comment (benper44) [reply ExtractArubaSW.ps1]: done

File: ExtractArubaSW.ps1 (hunk 6 above)
Comment (benper44) [reply ExtractArubaSW.ps1]: done

File: ExtractArubaSW.ps1 (hunk 7 above)
Comment (benper44) [reply ExtractArubaSW.ps1]: done

File: ExtractArubaSW.ps1 (hunk 8 above)
Comment (benper44) [reply ExtractArubaSW.ps1]: done

File: ExtractArubaSW.ps1 [hunk 9]
```diff
@@ -0,0 +1,584 @@
...
+$Column = 1
+$anum = 1
+$bnum = 1
+
+
+
+
+# Entering data in Excel
+
+ function ExcelFormatArray  {
```
Comment (alagoutte) [inline ExtractArubaSW.ps1]: can use Powershell style cmdlet (like Add-ExcelFormatArray) you can use Get-Verbs to get list of valid verb for cmdlet

Comment (benper44) [reply ExtractArubaSW.ps1]: done

File: ExtractArubaSW.ps1 [hunk 10]
```diff
@@ -0,0 +1,584 @@
...
+$Column = 1
+$anum = 1
+$bnum = 1
+
+
+
+
+# Entering data in Excel
+
+ function Add-ExcelFormatArray {
```
Comment (alagoutte) [inline ExtractArubaSW.ps1:361]: Can you move function on the top of file ?

File: ExtractArubaSW.ps1 [hunk 11]
```diff
@@ -0,0 +1,584 @@
...
+
+
+This script requires the PowerArubaSW module to be installed. More information at https://github.com/PowerAruba/PowerArubaSW
+This script uses Microsoft Excel, the software will need to be installed on the computer on which the script is running.
+
+  .EXAMPLE
+
+The script can be executed with the following argument list
+
+ExtractArubaSW.ps1 -CompanyName <company name> -IPSwitch <switch ip> -LoginSwitch <switch login> -PassSwitch <switch password>
```
Comment (alagoutte) [inline ExtractArubaSW.ps1:31]: can you rename (to Extract-Port-ArubSW) and move the script to a folder (Tools ? scripts ? examples ?)

File: ExtractArubaSW.ps1 [hunk 12]
```diff
@@ -0,0 +1,584 @@
...
+
+
+
+$row = 1
+$Column = 3
+$anum = 1
+$bnum = 1
+
+
+    Add-ExcelFormatArray -name $infoswitch.name -letter1 C -letter2 G -emplacementtxt xlLeft -fontcolorindex 1 -fontsize 10 -interiorcolorindex 19
```
Comment (alagoutte) [inline ExtractArubaSW.ps1:433]: fix indent (remove extra space)

and there is other issue after... (about indent and trailing whitespace)

File: ExtractArubaSW.ps1 [hunk 13]
```diff
@@ -0,0 +1,584 @@
...
+    
+    }
+
+
+
+
+# Saving the excel file:
+$workbook.SaveAs("$env:USERPROFILE\Desktop\$($date)_$($CompanyName)_$($ipswitch).xlsx")
+
+break
```
Comment (alagoutte) [inline ExtractArubaSW.ps1:583]: really needed ?

File: ExtractArubaSW.ps1 [hunk 14]
```diff
@@ -0,0 +1,584 @@
...
+                $countvlan ++
+                
+                }
+
+        $comacountvlan = 0
+      
+        foreach ($vlanp in $v) {
+
+            
+            if ($comacountvlan -eq $countvlan) {
```
Comment (alagoutte) [inline ExtractArubaSW.ps1:287]: i think i can be more easy for make this i will look after

File: ExtractArubaSW.ps1 [hunk 15]
```diff
@@ -0,0 +1,584 @@
...
+
+foreach ($port in $run) {
+ 
+    $v = $vlanports | Where-Object {$_.port_id -eq $port.id}
+    $tagged=$null
+    $w = $lldpremote | Where-Object {$lldpremote.local_port -eq $port.id} 
+
+    $countvlan = -1
+        
+        foreach ($vlanp in $v) {
```
Comment (alagoutte) [inline ExtractArubaSW.ps1:276]: i think there is a $v.count() ? for this

File: ExtractArubaSW.ps1 [hunk 16]
```diff
@@ -0,0 +1,584 @@
...
+
+        try {
+
+                Connect-ArubaSW -Server $ipswitch -Username $loginswitch -Password $mysecpassword
+                
+                }
+
+        Catch { 
+
+                cls
```
Comment (alagoutte) [inline ExtractArubaSW.ps1:189]: avoid alias use Clear-Host

**Review by alagoutte — CHANGES_REQUESTED** [review]

File: ExtractArubaSW.ps1 (hunk 10 above)
Comment (benper44) [reply ExtractArubaSW.ps1:361]: done

File: ExtractArubaSW.ps1 (hunk 11 above)
Comment (benper44) [reply ExtractArubaSW.ps1:31]: done

File: ExtractArubaSW.ps1 (hunk 12 above)
Comment (benper44) [reply ExtractArubaSW.ps1:433]: done

File: ExtractArubaSW.ps1 (hunk 13 above)
Comment (benper44) [reply ExtractArubaSW.ps1:583]: done

File: ExtractArubaSW.ps1 (hunk 16 above)
Comment (benper44) [reply ExtractArubaSW.ps1:189]: done

**Comment by alagoutte** [comment]
Thanks Benjamin !

**Pull request merged**